crates/bevy_math/src/common_traits.rs

```diff
@@ -47,29 +50,100 @@ pub trait VectorSpace:
     /// Note that the value of `t` is not clamped by this function, so extrapolating outside
     /// of the interval `[0,1]` is allowed.
     #[inline]
-    fn lerp(self, rhs: Self, t: f32) -> Self {
-        self * (1. - t) + rhs * t
+    fn lerp(self, rhs: Self, t: Self::Scalar) -> Self {
+        self * (Self::Scalar::ONE - t) + rhs * t
     }
 }
 
 impl VectorSpace for Vec4 {
+    type Scalar = f32;
     const ZERO: Self = Vec4::ZERO;
 }
 
 impl VectorSpace for Vec3 {
+    type Scalar = f32;
     const ZERO: Self = Vec3::ZERO;
 }
 
 impl VectorSpace for Vec3A {
+    type Scalar = f32;
     const ZERO: Self = Vec3A::ZERO;
 }
 
 impl VectorSpace for Vec2 {
+    type Scalar = f32;
     const ZERO: Self = Vec2::ZERO;
 }
 
-impl VectorSpace for f32 {
+impl VectorSpace for DVec4 {
+    type Scalar = f64;
+    const ZERO: Self = DVec4::ZERO;
+}
+
+impl VectorSpace for DVec3 {
+    type Scalar = f64;
+    const ZERO: Self = DVec3::ZERO;
+}
+
+impl VectorSpace for DVec2 {
+    type Scalar = f64;
+    const ZERO: Self = DVec2::ZERO;
+}
+
+// Every scalar field is a 1-dimensional vector space over itself.
+impl<T: ScalarField> VectorSpace for T {
+    type Scalar = Self;
+    const ZERO: Self = Self::ZERO;
+}
+
+/// A type that supports the operations of a scalar field. Every vector space is defined over a
```

Comment:
Not sure the mention of vector spaces is relevant, Fields are generally useful unto themselves.

Reply:
Ok, I'll move it to the associated type on `VectorSpace`.